### PR TITLE
fix for 0D variable pio interface

### PR DESCRIPTION
**Pull request by jedwards4b**
### 
remove start and count arguments from 0D variable interface to PIO

### 
This bug shows up when using PIO_VERSION=2 and DEBUG=TRUE

Contributors other than yourself, if any:

CTSM Issues Fixed (include github issue #):

Are answers expected to change (and if so in what way)? NO

Any User Interface Changes (namelist or namelist defaults changes)?

Testing performed, if any:
Tested with case f19_f19_mg17.I2000Clm50SpGs using both PIO versions and with and without DEBUG=TRUE

(List what testing you did to show your changes worked as expected)
(This can be manual testing or running of the different test suites)
(Documentation on system testing is here: https://github.com/ESCOMP/ctsm/wiki/System-Testing-Guide)
(aux_clm on cheyenne for intel/gnu and izumi for intel/gnu/nag/pgi is the standard for tags on master)

**NOTE: Be sure to check your Coding style against the standard:**
https://github.com/ESCOMP/ctsm/wiki/CTSM-coding-guidelines

### CODE REVIEW

File: src/main/ncdio_pio.F90.in [hunk 1]
```diff
@@ -1291,7 +1291,8 @@ contains
        else
           idata1d(1) = 0
        end if
-       status = pio_put_var(ncid, varid, start(1:1), count(1:1), idata1d)
+!       status = pio_put_var(ncid, varid, start(1:1), count(1:1), idata1d)
```
Comment (billsacks) [inline src/main/ncdio_pio.F90.in]: This line should be removed rather than commented out. I can do that with your permission, or you can do it.

**Comment by jedwards4b** [comment]
thanks, I've removed the line.

**Comment by ekluzek** [comment]
It sounds like this addresses #124.

**Comment by billsacks** [comment]
@jedwards4b - It turns out that this change is causing a lot of test failures. Although I haven't looked into the exact code that causes the problem, the problem generally seems to be when there is a vector of values on the netCDF file, but it is written one value at a time. With the new code, the first value gets set, and all the others remain 0.

From spot-checking a few of the test failures, I see this manifesting itself in two ways:

(1) Restart tests with more than one history file are failing with the following message in the lnd.log file upon restart:

```
 hist_restart_ncd ERROR: history_tape_in_use on restart file
 disagrees with current run: For tape            2
 On restart file:  F
 In current run :  T
 This suggests that this tape was empty in one case,
 but non-empty in the other. (history_tape_in_use .false.
 means that history tape is empty.)
 ENDRUN: ERROR in histFileMod.F90 at line 4024
 ERROR:
  ERROR: history_tape_in_use differs from restart file. You can NOT change histo
 ry options on restart.
```

Looking at the relevant variable on the clm2.r file, I see: `history_tape_in_use = 1, 0`, whereas I think this should be `1, 1` in this case.

An example failing test is `ERS_D_Ld3.f10_f10_musgs.I1850Clm50BgcCrop.cheyenne_intel.clm-default`

(2) When there are history files with multiple time slices on them, various date-related fields are incorrect. A test illustrating this issue is `SMS_Lm1_D.f10_f10_musgs.I1850Clm50BgcCrop.cheyenne_intel.clm-output_crop_highfreq`: You can compare it with baselines (in `/glade/p/cgd/tss/ctsm_baselines`), and you'll see that the h7 file has differences in the fields mcdate, mcsec, mdcur, mscur, nstep and time: With the new code, these are 0 for every time step after the first.

**Comment by billsacks** [comment]
@mvertens suggested a fix for this fix. I have implemented this in e414c6e2. This resolves the issues in the two tests mentioned above. I am now kicking off the full test suite.

**Comment by jedwards4b** [comment]
This looks good, thanks for figuring it out.

**Pull request merged**